src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
The `ChangesetCooker` is used by `Sync::Client` when applying remote changesets. Remote changesets are applied after new writelogs are received from the server. So if a `Sync::Client` still exists after we destroy the `SharedGroup` then there could be an issue. I'm not sure if this is the case though - I remember earlier object-store `SyncClients` were kept alive but I'm not sure if this is still the case.